test/lib/interpreters/split_clustering_test.rb

```diff
@@ -744,5 +747,50 @@ def test_clustering_with_sticky_vehicles
       assert_equal 2, (result[:routes].find{ |r| r[:vehicle_id] == 'vehicle_0' }[:activities].collect{ |a| a[:service_id] } & ['service_1', 'service_5']).size
       assert_equal 1, (result[:routes].find{ |r| r[:vehicle_id] == 'vehicle_1' }[:activities].collect{ |a| a[:service_id] } & ['service_12']).size
     end
+
+    def test_list_vehicles
+      # with timewindow
+      vrp = TestHelper.create(VRP.basic)
+      vrp.vehicles.first.timewindow = Models::Timewindow.new(start: 0, end: 10)
+      assert_equal 7, Interpreters::SplitClustering.list_vehicles({ start: 0, end: 6 }, vrp.vehicles).size
+      assert_equal 7, Interpreters::SplitClustering.list_vehicles({ start: 0, end: 10 }, vrp.vehicles).size
```

Comment:
If we have 11 days in the period, why do we only have 7 vehicles ?

Could you add some comments on what is expected and why ?

Reply:
_work_day_ partition provides one cluster per working day (monday, tuesday, ..., sunday), not for every day of schedule. This ensures that all visits of one service will take place at the same week day as other visits. That is why we do need to consider all visits in the same cluster.

I will add some comments in the test.